lineapy/plugins/taskgen.py

```diff
@@ -0,0 +1,221 @@
+                BaseSessionWriter().get_session_artifact_function_call_block(
+                    nc,
+                    source_module=f"{pipeline_name}_module",
+                )
+            )
+
+            task_def: TaskDefinition = TaskDefinition(
+                function_name=nc.safename,
+                user_input_variables=artifact_user_input_variables,
+                loaded_input_variables=loaded_input_vars,
+                typing_blocks=user_input_var_typing_block,
+                call_block=function_call_block,
+                return_vars=nc.return_variables,
+                pipeline_name=pipeline_name,
+            )
+            task_definitions[nc.safename] = task_def
+
+    return task_definitions
+
+
+def get_session_task_definition(
+    artifact_collection: ArtifactCollection, pipeline_name: str
+) -> Dict[str, TaskDefinition]:
+    """
+    get_session_task_definition returns a task definition for each session in the pipeline.
+    """
+    task_definitions: Dict[str, TaskDefinition] = dict()
+
+    for session_artifacts in artifact_collection.sort_session_artifacts():
+
+        session_input_parameters_spec = (
+            BaseSessionWriter().get_session_input_parameters_spec(
+                session_artifacts
+            )
+        )
+        session_input_variables = list(session_input_parameters_spec.keys())
+        user_input_var_typing_block = [
+            f"{var} = {session_input_parameters_spec[var].value_type}({var})"
+            for var in session_input_variables
+        ]
+
+        raw_function_call_block = (
+            BaseSessionWriter().get_session_function_callblock(
+                session_artifacts
+            )
+        )
+        function_call_block = (
+            f"artifacts = {pipeline_name}_module.{raw_function_call_block}"
+        )
+        return_vars = [
+            nc.safename
+            for nc in session_artifacts.usercode_nodecollections
+            if isinstance(nc, ArtifactNodeCollection)
+        ]
+
+        function_name = BaseSessionWriter().get_session_function_name(
+            session_artifacts
+        )
+
+        task_def: TaskDefinition = TaskDefinition(
+            function_name=function_name,
+            user_input_variables=session_input_variables,
+            loaded_input_variables=[],
+            typing_blocks=user_input_var_typing_block,
+            call_block=function_call_block,
+            return_vars=return_vars,
+            pipeline_name=pipeline_name,
+        )
+
+        task_definitions[function_name] = task_def
+
+    return task_definitions
+
+
+def get_allsessions_task_definition(
+    pipeline_name: str,
+) -> Dict[str, TaskDefinition]:
+    """
+    get_allsessions_task_definition returns a single task definition for the whole pipeline.
+    """
+    indentation_block = " " * 4
+    return {
+        "run_all": TaskDefinition(
+            function_name="run_all",
+            user_input_variables=[],
```

Comment:
Not allowed user input parameters here?

Reply:
i think this is a mistake on my part, let me revise this

Reply:
fixed input variables and return variable